erc20-migration/hardhat.config.js

```diff
@@ -183,18 +207,146 @@ task("contractSetup", "To be used just for testing", async (taskArgs, hre) => {
   let daoVestingBeneficiary = await daoVesting.beneficiary();
-  if (daoVestingBeneficiary != process.env.HORIZEN_DAO){
-    console.error("Wrong beneficiary for Horizen DAO Vesting Contract. Expected: " +  process.env.HORIZEN_DAO + ", found: " + daoVestingBeneficiary);
+  if (daoVestingBeneficiary != process.env.HORIZEN_DAO) {
+    console.error("Wrong beneficiary for Horizen DAO Vesting Contract. Expected: " + process.env.HORIZEN_DAO + ", found: " + daoVestingBeneficiary);
     exit(-1);
   }
 
+
+  if (taskArgs.verify) {
+    //Contract verification
+    console.log(`Starting contract verification`);
+    try {
+      await hre.run("verify:verify", {
+        address: await ZenMigrationFactory.getAddress(),
+        constructorArguments: [
+          admin.address
+        ],
+      });
+    }
+    catch (error) {
+      if (error.name != "ContractAlreadyVerifiedError") {
+        console.log("Error verifying Migration factory contract");
+        console.log(error);
+        exit(-1);
+      }
+    }
+    console.log(`Migration factory contract verified`);
+
+
+    try {
+      await hre.run("verify:verify", {
+        address: await ZenMigrationFactory.zendVault(),
+        constructorArguments: [
+          admin.address,
+          base_message
+        ],
+      });
+    }
+    catch (error) {
+      if (error.name != "ContractAlreadyVerifiedError") {
+        console.log("Error verifying Zend vault contract");
+        console.log(error);
+        exit(-1);
+      }
+
+    }
+    console.log(`Zend vault contract verified`);
+
+
+    try {
+      await hre.run("verify:verify", {
+        address: await ZenMigrationFactory.eonVault(),
+        constructorArguments: [
+          admin.address
+        ],
+      });
+    }
+    catch (error) {
+      if (error.name != "ContractAlreadyVerifiedError") {
+        console.log("Error verifying Eon vault contract");
+        console.log(error);
+        exit(-1);
+      }
+
+    }
+    console.log(`Eon vault contract verified`);
+ 
+    try {
+      await hre.run("verify:verify", {
+        address: await ZenMigrationFactory.token(),
+        constructorArguments: [
+          tokenName,
+          tokenSymbol,
+          await ZenMigrationFactory.eonVault(),
+          await ZenMigrationFactory.zendVault(),
+          await ZenMigrationFactory.horizenFoundationVestingContract(),
+          await ZenMigrationFactory.horizenDaoVestingContract()
+        ],
+      });
+    }
+    catch (error) {
+      if (error.name != "ContractAlreadyVerifiedError") {
+        console.log("Error verifying ERC20 contract");
+        console.log(error);
+        exit(-1);
+      }
+
+    }
+    console.log(`ERC20 contract verified`);
+
+    const VESTING_TIME_BETWEEN_INTERVALS = 30 * 24 * 60 * 60; //Length in seconds of each vesting interval (30 days)
+    const VESTING_INTERVALS = 48; //Total number of vesting intervals
```

Comment:
instead of hardcode can be obtained by the smart contract: public fields  timeBetweenClaims  and intervalsToClaim

Reply:
Good idea! Done